server/auth/privileges.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+// Privilege represents some permission for a database object.
+// https://www.postgresql.org/docs/15/ddl-priv.html
+type Privilege string
+
+const (
+	Privilege_SELECT       = "r"
+	Privilege_INSERT       = "a"
+	Privilege_UPDATE       = "w"
+	Privilege_DELETE       = "d"
+	Privilege_TRUNCATE     = "D"
+	Privilege_REFERENCES   = "x"
+	Privilege_TRIGGER      = "t"
+	Privilege_CREATE       = "C"
+	Privilege_CONNECT      = "c"
+	Privilege_TEMPORARY    = "T"
+	Privilege_EXECUTE      = "X"
+	Privilege_USAGE        = "U"
+	Privilege_SET          = "s"
+	Privilege_ALTER_SYSTEM = "A"
+)
+
+// PrivilegeObject is the database object that privileges are applied to.
+// https://www.postgresql.org/docs/15/ddl-priv.html
+type PrivilegeObject byte
```

Comment:
`PrivilegeTarget` or `PrivilegeTargetType` would be clearer

Reply:
I think they're referred to as "objects" specifically, but I'll see if there's other terminology used that may be clearer.